### PR TITLE
[pylint] alias checker and tests

### PR DESCRIPTION
**Pull request by l0lawrence**
This will affect MetricsAdvisor and KeyVault Secrets and Search 

Some changes to README links, other PR will fix link errors

### CODE REVIEW

File: scripts/pylint_custom_plugin/pylint_guidelines_checker.py [hunk 1]
```diff
@@ -1706,6 +1706,67 @@ def visit_functiondef(self, node):
     visit_asyncfunctiondef = visit_functiondef
 
 
+class CheckNoAliasGeneratedCode(BaseChecker):
```
Comment (l0lawrence) [inline scripts/pylint_custom_plugin/pylint_guidelines_checker.py:1709]: Naming?

File: scripts/pylint_custom_plugin/pylint_guidelines_checker.py [hunk 2]
```diff
@@ -1706,6 +1706,68 @@ def visit_functiondef(self, node):
     visit_asyncfunctiondef = visit_functiondef
 
 
+class CheckNamingMismatch(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    name = "check-naming"
```
Comment (catalinaperalta) [inline scripts/pylint_custom_plugin/pylint_guidelines_checker.py:1712]: What is this name used for? And should it be more explicit? Like `check-naming-mismatch`

File: scripts/pylint_custom_plugin/README.md [hunk 3]
```diff
@@ -46,16 +46,17 @@ In the case of a false positive, use the disable command to remove the pylint er
...
\ No newline at end of file
+| docstring-admonition-needs-newline                 | Add a blank newline above the .. literalinclude statement.                                                                                                           | # pylint:disable=docstring-admonition-needs-newline                                      | No guideline, just helps our docs get built correctly for microsoft docs.                     |
+| aliasing-generated-code                 | Do not alias models imported from the generated code.                                                                       | # pylint:disable=aliasing-generated-code                                      | [link](https://github.com/Azure/autorest/blob/main/docs/generate/built-in-directives.md)                     |
```
Comment (catalinaperalta) [inline scripts/pylint_custom_plugin/README.md:62]: The disable value needs to be updated.

File: scripts/pylint_custom_plugin/pylint_guidelines_checker.py [hunk 4]
```diff
@@ -1706,6 +1706,68 @@ def visit_functiondef(self, node):
...
+    priority = -1
+    msgs = {
+        "C4745": (
+            "Exposing aliased generated code."
```
Comment (catalinaperalta) [inline scripts/pylint_custom_plugin/pylint_guidelines_checker.py:1716]: Does this line need to tell people NOT to expose aliased generated code?

**Review by catalinaperalta — COMMENTED** [review]
This is looking great! Left some questions and comments.

File: scripts/pylint_custom_plugin/pylint_guidelines_checker.py (hunk 4 above)
Comment (l0lawrence) [reply scripts/pylint_custom_plugin/pylint_guidelines_checker.py]: I changed it to say do not alias generated code

**Review by catalinaperalta — APPROVED** [review]
LGTM

**Pull request merged**